src/transformers/generation/tf_utils.py

```diff
@@ -1500,17 +1382,22 @@ def greedy_search(
         # 1. init greedy_search values
         logits_processor = logits_processor if logits_processor is not None else TFLogitsProcessorList()
 
-        max_length = max_length if max_length is not None else self.config.max_length
-        pad_token_id = pad_token_id if pad_token_id is not None else self.config.pad_token_id
-        eos_token_id = eos_token_id if eos_token_id is not None else self.config.eos_token_id
-        output_scores = output_scores if output_scores is not None else self.config.output_scores
-        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        max_length = max_length if max_length is not None else self.generation_config.max_length
+        pad_token_id = pad_token_id if pad_token_id is not None else self.generation_config.pad_token_id
+        eos_token_id = eos_token_id if eos_token_id is not None else self.generation_config.eos_token_id
+        output_scores = output_scores if output_scores is not None else self.generation_config.output_scores
+        output_attentions = (
+            output_attentions if output_attentions is not None else self.generation_config.output_attentions
+        )
         output_hidden_states = (
-            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+            output_hidden_states if output_hidden_states is not None else self.generation_config.output_hidden_states
         )
         return_dict_in_generate = (
-            return_dict_in_generate if return_dict_in_generate is not None else self.config.return_dict_in_generate
+            return_dict_in_generate
+            if return_dict_in_generate is not None
+            else self.generation_config.return_dict_in_generate
         )
+        use_cache = model_kwargs.pop("use_cache", self.generation_config.use_cache)
```

Comment:
This pattern, pulling `use_cache` out of the `model_kwargs`, is the only “new” modification when compared to the PR made on the PT side.

It was also an existing bug that went undetected: when `use_cache` is set in `model_kwargs`, the XLA-compiled loop converts it to a `tf.bool` variable (because it is part of an input/output variable in the `tf.while_loop`, and therefore assumed as dynamic at runtime). Because it is converted to a `tf.bool`, XLA model compilation fails due to lines like `present_key_values = () if use_cache else None`, which require `use_cache` to be a static variable.

Workaround: pull `use_cache` out of the `model_kwargs` and use the corresponding static variable in the `tf.while_loop`